diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index dcbf7ab..9cad45f 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -63,6 +63,10 @@ elseif(NOT EPROSIMA_INSTALLER)
 	$<$<STREQUAL:$<TARGET_PROPERTY:${PROJECT_NAME},TYPE>,SHARED_LIBRARY>:${PROJECT_NAME_UPPER}_DYN_LINK>
         )
 
+    if(QNX)
+        target_compile_definitions(${PROJECT_NAME} PRIVATE _QNX_SOURCE)
+    endif()
+
     # Define public headers
     target_include_directories(${PROJECT_NAME} PUBLIC
         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
