diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 3b8f886..2d44055 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -63,6 +63,10 @@ target_compile_definitions(${PROJECT_NAME}
     $<$<TARGET_EXISTS:${PROJECT_NAME}>:$<$<STREQUAL:$<TARGET_PROPERTY:${PROJECT_NAME},TYPE>,SHARED_LIBRARY>:${PROJECT_NAME_UPPER}_DYN_LINK>>
     )
 
+if(QNX)
+    target_compile_definitions(${PROJECT_NAME} PRIVATE _QNX_SOURCE)
+endif()
+
 # Define public headers
 target_include_directories(${PROJECT_NAME} PUBLIC
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
